### DISCUSSION TITLE
Processing 3 coding error

### DISCUSSION THREAD

**Discussion by GoodByeWorldV** [community/community#126736] [Programming Help] [answered]
### Select Topic Area

Question

### Body

First, my English skill is broken. So my question makes you confuse. I apology that...

I use 'Processing 3'. And I made code to create 100 sheep to list []. Than I choose one sheep to mouse, that sheep follow my mouse... But all sheep follow my mouse... How can I choose just one sheep??

void mousePressed(){
  for(int k = 0; k<100; k++){
    float d = dist(mouseX, mouseY, x[k], y[k]);
    if((d < x[k]+7*Sheep_unit || d > x[k]-7*Sheep_unit) && (d < y[k]+8*Sheep_unit || d > y[k]-8*Sheep_unit)){
      select = k;
    }
    num_size[select] = 20;
    num_color[select] = color(255, 0, 0);
  }
}

^^^ This is my code, the sheep create to coordinate (x[k], y[k]). x[] and y[] have each coordinate by random. + And each sheep have number 1-100. If they choose, the number bigger and be red. num_size[] and num_color[] make that.

I don't know how to choose JUST one sheep... please help me ;(

**Comment by GoodByeWorldV** [comment]
and I wonder how to upgrade my coding skill...

**Comment by CopperEagle** [comment, marked as answer]
Hey there @GoodByeWorldV

Note that you make the assignments to `num_size` and  `num_color` inside the loop. So you may select many sheep. Likely what you want to do is to loop until you are in the `if` branch, setting `select = k;`. Then you probably want to `break;` from the loop and *after that*, outside the loop, do the two assignments `num_size[select] = 20;`.  Like this:

```java
void mousePressed(){
    for(int k = 0; k<100; k++){
        float d = dist(mouseX, mouseY, x[k], y[k]);
        if(...){
            select = k;
            break;
        }
    }
    num_size[select] = 20;
    num_color[select] = color(255, 0, 0);
}

```

Does this solve the issue?

**Reply by GoodByeWorldV** [reply]
Oh why didn't I come up with 'break;'? @CopperEagle thank you!!!
